src/composables/useMessageInfo.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { computed, unref, ref } from 'vue'
+
+import { getCapabilities } from '@nextcloud/capabilities'
+import moment from '@nextcloud/moment'
+
+import { useConversationInfo } from './useConversationInfo.js'
+import { useStore } from './useStore.js'
+
+const canDeleteMessageUnlimited = getCapabilities()?.spreed?.features?.includes('delete-messages-unlimited')
+const canEditMessage = getCapabilities()?.spreed?.features?.includes('edit-messages')
+
+/**
+ * Check whether the user can edit the message or not
+ *
+ * @param {import('vue').Ref} tokenRef conversation token
+ * @param {import('vue').Ref} messageIdRef message id to edit
+ * @return {import('vue').ComputedRef<boolean>}
+ */
+export function useMessageInfo(tokenRef = ref(null), messageIdRef = ref(null)) {
```

Comment:
Cases where [token, id, conversation, message] are `undefined` and `null` are not handled, right? :eyes: